src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -0,0 +1,232 @@
+
+		return config;
+	}
+
+	private alertUser(tsconfigPath: string): void {
+		const event = {
+			messageKey: 'info.usingEngineConfigFile', args: [tsconfigPath], type: 'INFO', handler: 'UX', verbose: true, time: Date.now()
+		};
+		this.outputProcessor.emitEvents([event]);
+	}
+
+	async findTsconfig(target: string): Promise<string> {
+		let tsconfigPath: string;
+
+		// Check config
+		tsconfigPath = await this.getTsconfigFromConfig();
+
+		// Check target
+		if (!tsconfigPath) {
+			tsconfigPath = await this.getTsconfigFromTarget(target);
+		}
+
+		// Check current working directory
+		if (!tsconfigPath) {
+			tsconfigPath = await this.checkDirectoryForTsconfig('');
+		}
+
+		this.logger.trace(`Using ${TS_CONFIG} from ${tsconfigPath}`);
+		return tsconfigPath;
+	}
+
+	private async getTsconfigFromConfig(): Promise<string> {
+		let tsconfigPath = '';
+		// Check if overriddenConfig is available
+		const overriddenConfig = this.config.getOverriddenConfigPath(this.getName());
+		if (!overriddenConfig) {
+			this.logger.trace(`Did not find an overridden path from Config`);
+			return tsconfigPath;
+		}
+
+		// Complain if overriddenConfig does not exist.
+		if (!(await this.fileHandler.exists(path.resolve(overriddenConfig)))) {
+			throw new SfdxError(`Invalid path in Config: ${overriddenConfig}`); // FIXME: move to messages.js
+		}
+
+		// If overriddenConfig is a directory, look for tsconfig.json file.
+		if (await this.fileHandler.isDir(overriddenConfig)) {
+			this.logger.trace(`Looking for ${TS_CONFIG} inside overridden dir from config ${overriddenConfig}`);
+			tsconfigPath = await this.checkDirectoryForTsconfig(overriddenConfig);
+			// Complain if we can't find a tsconfig.json inside.
+			if (!tsconfigPath) {
+				throw new SfdxError(`Overridden path [${overriddenConfig}] does not contain ${TS_CONFIG}`);
+			}
+		} else if (overriddenConfig.endsWith(TS_CONFIG)) { // If overriddenConfig is a file, check if the file is tsconfig.json
+			this.logger.trace(`Found ${TS_CONFIG} directly as overridden path in Config, ${overriddenConfig}`);
+			tsconfigPath = overriddenConfig;
+		} else { // Complain if the file is something else
+			throw new SfdxError(`Cannot find ${TS_CONFIG} with overridden path ${overriddenConfig}`);
+		}
+
+		return tsconfigPath;
+	}
+
+	private async getTsconfigFromTarget(target: string): Promise<string> {
+		let tsconfigPath = '';
+		if (await this.fileHandler.isDir(target)) {
+			tsconfigPath = await this.checkDirectoryForTsconfig(target);
+		}
+		// TODO: if this is a pattern, should we look at each directory from bottom to top?
+		return tsconfigPath;
+	}
+
+	private async checkDirectoryForTsconfig(givenPath: string): Promise<string> {
+		if (await this.fileHandler.exists(path.resolve(givenPath, TS_CONFIG))) {
+			const tsconfigPath = path.resolve(givenPath, TS_CONFIG);
+			this.logger.trace(`Found ${TS_CONFIG} in directory ${givenPath}`);
+			return tsconfigPath;
+		}
+		return '';
```

Comment:
Could this return null instead of empty string? If '' is required, maybe declare it as a global constant, I was trying to read through the code and see if it would ever be interpreted different than null, but I couldn't find anything.

Reply:
Let me define a global value.